gnd/src/androidTest/java/com/google/android/gnd/AddFeatureTest.java

```diff
@@ -0,0 +1,152 @@
+import static androidx.test.espresso.Espresso.onData;
+import static androidx.test.espresso.Espresso.onView;
+import static androidx.test.espresso.action.ViewActions.click;
+import static androidx.test.espresso.assertion.ViewAssertions.matches;
+import static androidx.test.espresso.matcher.ViewMatchers.isCompletelyDisplayed;
+import static androidx.test.espresso.matcher.ViewMatchers.withId;
+import static androidx.test.espresso.matcher.ViewMatchers.withText;
+import static org.hamcrest.CoreMatchers.allOf;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.Matchers.instanceOf;
+import static org.hamcrest.Matchers.not;
+
+import android.content.Context;
+import androidx.room.Room;
+import androidx.test.core.app.ActivityScenario;
+import androidx.test.ext.junit.rules.ActivityScenarioRule;
+import com.google.android.gnd.persistence.local.LocalDatabaseModule;
+import com.google.android.gnd.persistence.local.room.LocalDatabase;
+import com.google.android.gnd.persistence.remote.FakeRemoteDataStore;
+import com.google.android.gnd.persistence.remote.FakeRemoteStorageManager;
+import com.google.android.gnd.persistence.remote.RemoteDataStore;
+import com.google.android.gnd.persistence.remote.RemoteStorageManager;
+import com.google.android.gnd.persistence.remote.RemoteStorageModule;
+import com.google.android.gnd.persistence.uuid.FakeUuidGenerator;
+import com.google.android.gnd.persistence.uuid.OfflineUuidGenerator;
+import com.google.android.gnd.system.auth.AuthenticationManager;
+import com.google.android.gnd.system.auth.AuthenticationModule;
+import com.google.android.gnd.system.auth.FakeAuthenticationManager;
+import dagger.Binds;
+import dagger.Module;
+import dagger.Provides;
+import dagger.hilt.InstallIn;
+import dagger.hilt.android.components.ApplicationComponent;
+import dagger.hilt.android.qualifiers.ApplicationContext;
+import dagger.hilt.android.testing.HiltAndroidRule;
+import dagger.hilt.android.testing.HiltAndroidTest;
+import dagger.hilt.android.testing.UninstallModules;
+import javax.inject.Singleton;
+import org.junit.Rule;
+import org.junit.Test;
+
+@UninstallModules({AuthenticationModule.class, RemoteStorageModule.class,
+  LocalDatabaseModule.class})
+@HiltAndroidTest
+public class AddFeatureTest {
+
+  // Ensures that the Hilt component is initialized before running the ActivityScenarioRule.
+  public @Rule(order = 0)
+  HiltAndroidRule hiltRule = new HiltAndroidRule(this);
+
+  // Sets the preferences so no login is required and an active project is selected.
+  public @Rule(order = 1)
+  SetPreferencesRule preferencesRule = new SetPreferencesRule();
+
+  public @Rule(order = 2)
+  ActivityScenarioRule scenarioRule =
+    new ActivityScenarioRule(MainActivity.class);
+
+  @Module
+  @InstallIn(ApplicationComponent.class)
+  abstract class TestDependenciesModule {
+
+    @Binds
+    @Singleton
+    abstract AuthenticationManager bindAuthenticationManager(
+      FakeAuthenticationManager authenticationManager);
```

Comment:
Close parenthesis at the same level as the first character of the line that opens it. Not a hard rule but I think it's the trend now. 

To avoid bikeshedding like this I recommend you install spotless. It's 10 lines in your build.gradle file that will save a lot of time.

Reply:
We're already using `checkstyle` with the google java style guide, however, there is no rule for this. I tried trawling through the docs to see if it's easy to add, but tbh after 15 mins of researching I didn't think it was worth it.

Reply:
@gino-m Any thoughts on this? If it's an easy add it's probably worth adding a rule for this, just for code consistency.